### DISCUSSION TITLE
About the input training dataset

### DISCUSSION THREAD

**Discussion by 0WellyWang0** [mir-group/allegro#13] [Q&A] [answered]
Hi. I have a question about the dataset for training. For example, if I would like to find a "universal" potential of Fe and O system, could I combine and train the trajectory of Fe3O4.xyz, FeO.xyz, and Fe2O3.xyz together and finally produce a single result? I got problems when doing this kind of work: 1000 data points from Fe3O4.xyz + 1000 from FeO.xyz + 1000 from Fe2O3.xyz led to a combined.xyz file, but allegro only took the first 1000 data points from Fe3O4.xyz to train. Thanks.

**Comment by Linux-cpp-lisp** [comment, marked as answer]
Hi @0WellyWang0 ,

The files you provide determine what data is available to pick from, but you still need to set `n_train` and `n_val` to control how many data points are taken for the training and validation sets respectively.

**Reply by 0WellyWang0** [reply]
Thanks. Just a more specific question on the code: since FeOtraj.xyz and Fe2O3traj.xyz contained different compositions of Fe and O, shall I load them separately with 2 command sets and set n_train and n_val to train together like the following?

```
# -- data --
dataset: ase                                                                   
dataset_file_name:     FeOtraj.xyz                #1000 data points
ase_args:
  format: extxyz

dataset: ase                                                                   
dataset_file_name:     Fe2O3traj.xyz                #1000 data points
ase_args:
  format: extxyz

# training
n_train: 1600
n_val: 200
```

**Reply by Linux-cpp-lisp** [reply]
No, you should merge all your training data into one XYZ file.

It _is_ possible to use separate dataset definitions for the training dataset and validation dataset:
```
dataset: ase                                                                   
dataset_file_name:     FeOtraj.xyz                #1000 data points
ase_args:
  format: extxyz
                                                                  
validation_dataset_file_name:     somethingelse.xyz                #1000 data points
```

**Reply by 0WellyWang0** [reply]
I got the point and thanks a lot. Good luck with your work.
